src/main/java/spireMapOverhaul/zones/packattack/events/PackShrine.java

```diff
@@ -0,0 +1,195 @@
+package spireMapOverhaul.zones.packattack.events;
+
+import basemod.ReflectionHacks;
+import com.badlogic.gdx.math.MathUtils;
+import com.evacipated.cardcrawl.modthespire.Loader;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.cards.CardGroup;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.core.Settings;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.events.AbstractImageEvent;
+import com.megacrit.cardcrawl.localization.EventStrings;
+import com.megacrit.cardcrawl.relics.AbstractRelic;
+import com.megacrit.cardcrawl.unlock.UnlockTracker;
+import com.megacrit.cardcrawl.vfx.cardManip.PurgeCardEffect;
+import com.megacrit.cardcrawl.vfx.cardManip.ShowCardAndObtainEffect;
+import spireMapOverhaul.SpireAnniversary6Mod;
+import spireMapOverhaul.zones.packattack.PackAttackZone;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class PackShrine extends AbstractImageEvent {
+    public static final String ID = SpireAnniversary6Mod.makeID("PackShrine");
+    private static final EventStrings eventStrings = CardCrawlGame.languagePack.getEventString(ID);
+    private static final String NAME = eventStrings.NAME;
+    private static final String[] DESCRIPTIONS = eventStrings.DESCRIPTIONS;
+    private static final String[] OPTIONS = eventStrings.OPTIONS;
+    public static final String IMG_1 = SpireAnniversary6Mod.makeImagePath("events/PackAttack/PackShrine1.png");
+    public static final String IMG_2 = SpireAnniversary6Mod.makeImagePath("events/PackAttack/PackShrine2.png");
+    private static final int CARDS = 20;
+    private static final int COMMON_GOLD = 30;
+    private static final int A15_COMMON_GOLD = 20;
+    private static final int UNCOMMON_GOLD = 60;
+    private static final int A15_UNCOMMON_GOLD = 50;
+    private static final int RARE_GOLD = 120;
+    private static final int A15_RARE_GOLD = 100;
+
+    private int commonGold;
+    private int uncommonGold;
+    private int rareGold;
+    private boolean pickCard = false;
+    private boolean remove = false;
+
+    private int screenNum = 0;
+
+    public PackShrine() {
+        super(NAME, DESCRIPTIONS[0], IMG_1);
+        if (MathUtils.randomBoolean()) {
+            this.imageEventText.loadImage(IMG_2);
+        }
+
+        if (!Loader.isModLoaded("anniv5")) {
+            throw new RuntimeException("This event requires Packmaster to be loaded");
```

Comment:
Hmm, not sure if this is a good idea. If some mod does something weird with a dynamic way to get all events they might run into this accidentally.

Could we just set the only dialogue option to leave if that's the case instead?

Reply:
Good suggestion, that will be easy to do.